app/services/statistics/base.rb

```diff
@@ -3,11 +3,13 @@ class Base
     attr_reader :title, :count_by
 
     def initialize(options = {})
-      @id       = options[:id]
-      @title    = options[:title]
-      @search   = options[:search]
-      @url      = options[:url]
+      @id = options[:id]
+      @title = options[:title]
+      @search = options[:search]
+      @url = build_url(options[:url], options[:organization_id], options[:location_id])
```

Comment:
I don't see anywhere we actually provide `:url` in the options, perhaps we should just pass the parameters in the url directly? at a higher point in the stack we can actually use `statistics_url(id: id, organization_id: Organization.current.try(:id), location_id: Location.current.try(:id)` to generate the url using rails helpers instead of building it ourselves.

Reply:
Higher in the stack would mean somewhere where we have view context, the list of charts is in Statistics module though. I can use url_helpers since we don't need to specify host, so we can use `statistics_path` instead of `statistics_url`. But `statistics_path` does not accept additional parameters so I needed to use `url_for`. That seems to work but it is quite a change, therefore I'm keeping it in separate in case we want to keep this patch small, e.g. for backporting to older version. Specifying url option is now required.